libcloud/dns/drivers/dnspod.py

```diff
@@ -0,0 +1,345 @@
+        except DNSPodException:
+            e = sys.exc_info()[1]
+            if e.message == 'No domains':
+                return []
+        zones = self._to_zones(items=response.parse_body()['domains'])
+
+        return zones
+
+    def delete_zone(self, zone):
+        """
+        :param zone: Zone to be deleted.
+        :type zone: :class:`Zone`
+
+        :return: Boolean
+        """
+        action = '/Domain.Remove'
+        data = {'user_token': self.key, 'format': 'json', 'domain_id': zone.id}
+        data = urlencode(data)
+
+        try:
+            self.connection.request(action=action, method='POST',
+                                    data=data)
+        except DNSPodException:
+            e = sys.exc_info()[1]
+            if e.message == 'Domain id invalid':
+                raise ZoneDoesNotExistError(value=e.message, driver=self,
+                                            zone_id=zone.id)
+            else:
+                raise e
+
+        return True
+
+    def get_zone(self, zone_id):
+        """
+        :param zone_id: Zone domain name (e.g. example.com)
+        :return: :class:`Zone`
+        """
+        action = '/Domain.Info'
+        data = {'user_token': self.key, 'format': 'json', 'domain_id': zone_id}
+        data = urlencode(data)
+
+        try:
+            response = self.connection.request(action=action, method='POST',
+                                               data=data)
+        except DNSPodException:
+            e = sys.exc_info()[1]
+            if e.message in ['Domain not under you or your user',
+                             'Domain id invalid']:
+                raise ZoneDoesNotExistError(value=e.message, driver=self,
+                                            zone_id=zone_id)
+            else:
+                raise e
+        zone = self._to_zone(response.parse_body()['domain'])
+
+        return zone
+
+    def create_zone(self, domain, type='master', ttl=None, extra=None):
+        """
+        :param domain: Zone domain name (e.g. example.com)
+        :type domain: ``str``
+
+        :param type: Zone type (This is not really used. See API docs for extra
+          parameters)
+        :type type: ``str``
+
+        :param ttl: TTL for new records (This is used through the extra param)
+        :type ttl: ``int``
+
+        :param extra: Extra attributes that are specific to the driver
+        such as ttl.
+        :type extra: ``dict``
+
+        :rtype: :class:`Zone`
+        """
+        action = '/Domain.Create'
+        data = {'user_token': self.key, 'format': 'json', 'domain': domain}
+        if extra is not None:
+            data.update(extra)
+        data = urlencode(data)
+        try:
+            response = self.connection.request(action=action, method='POST',
+                                               data=data)
+        except DNSPodException:
+            e = sys.exc_info()[1]
+            if e.message in ['Domain is exists',
```

Comment:
You could save those messages in a constant, e.g. `ZONE_ALREADY_EXISTS_ERROR_MSGS = [ ... ]`.